### PR TITLE
Retry tar operations after 'Unexpected EOF in archive' during node setup

### PR DESCRIPTION
**Pull request by maddieford**
## Description

Issue # 

Opened this PR to address the following issue we're seeing on LISA runs where tarball operations fails on test node during setup

stderr:
tar: Unexpected EOF in archive
tar: Unexpected EOF in archive
tar: Error is not recoverable: exiting now
---

### PR information
- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/WALinuxAgent/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests_e2e/orchestrator/lib/agent_test_suite.py [hunk 1]
```diff
@@ -347,6 +348,23 @@ def _setup_node(self, install_test_agent: bool) -> None:
         log.info("Copying %s to %s:%s", agent_package_path, self.context.node.name, target_path)
         self.context.ssh_client.copy_to_node(agent_package_path, target_path)
 
+        #
```
Comment (narrieta) [inline tests_e2e/orchestrator/lib/agent_test_suite.py:351]: it's probably be simpler to just retry the copy when the stderr of the first command contains "tar: Unexpected EOF in archive"

File: tests_e2e/orchestrator/lib/agent_test_suite.py [hunk 2]
```diff
@@ -335,24 +312,70 @@ def _setup_node(self, install_test_agent: bool) -> None:
...
+                # Tarball creation and extraction was successful - no need to retry
+                tar_retries = 0
+
+            except CommandError as error:
+                if "tar: Unexpected EOF in archive" in error.stderr:
+                    tar_retries -= 1
+                    # Log the error with traceback to see which tar operation failed
+                    log.info(f"Tarball creation or extraction failed: \n{error}")
+                    # Retry tar operations
+                    if tar_retries > 0:
```
Comment (nagworld9) [inline tests_e2e/orchestrator/lib/agent_test_suite.py:375]: when we do retry, it's better to cleanup partial tar file if already created during previous attempts

Comment (narrieta) [reply tests_e2e/orchestrator/lib/agent_test_suite.py:375]: just as a comment, the first tar does "cvf".  "c" - creates the file (overwrites any existing file)

Comment (maddieford) [reply tests_e2e/orchestrator/lib/agent_test_suite.py:375]: tar cvf and scp both overwrite any existing files

Comment (nagworld9) [reply tests_e2e/orchestrator/lib/agent_test_suite.py:375]: good to know

**Review by nagworld9 — APPROVED** [review]

**Pull request merged**